harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -147,31 +153,48 @@ def download(self, path: Optional[str] = None) -> str:
                 raise NotImplementedError("Non-training checkpoints cannot be downloaded")
 
             checkpoint_storage = self.training.experiment_config["checkpoint_storage"]
-            if checkpoint_storage["type"] == "shared_fs":
-                src_ckpt_dir = self._find_shared_fs_path(checkpoint_storage)
-                shutil.copytree(str(src_ckpt_dir), str(local_ckpt_dir))
-            else:
-                local_ckpt_dir.mkdir(parents=True, exist_ok=True)
-                manager = storage.build(
-                    checkpoint_storage,
-                    container_path=None,
-                )
-                if not isinstance(
-                    manager,
-                    (
-                        storage.S3StorageManager,
-                        storage.GCSStorageManager,
-                        storage.AzureStorageManager,
-                    ),
-                ):
-                    raise AssertionError(
-                        "Downloading from Azure, S3 or GCS requires the experiment to be "
-                        "configured with Azure, S3 or GCS checkpointing, {} found instead".format(
-                            checkpoint_storage["type"]
+            try:
+                if via_master:
+                    if checkpoint_storage["type"] != "s3":
+                        raise NotImplementedError(
```

Comment:
Can we make this a `NoCheckpointDownloadViaMaster` or something like that (but better named)?

I don't see why we need to distinguish between the "s3 is configured but master doesn't have permissions" vs the "gcs is configured and master doesn't support that yet" cases.  From the user perspective, both of those mean "I have to figure out my own direct access", so I think they should be the same error.

Reply:
Or I don't check it here and rely on the API error, which provides similar info.